File: Civitas_TUI/src/civitas/Casilla_Calle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author alejandro
 */
public class Casilla_Calle extends Casilla {
    private TituloPropiedad tituloPropiedad;
    private float importe;
    
    Casilla_Calle(TituloPropiedad titulo){
        super(titulo.getNombre());
        tituloPropiedad = titulo;
        importe = titulo.getPrecioCompra();
    }
    
    @Override
    void recibeJugador(int iactual, ArrayList<Jugador> todos){
        if(super.jugadorCorrecto(iactual, todos)){
            informe(iactual,todos);
            Jugador jugador = todos.get(iactual);
            jugador.setPuedeComprar(true);

            
            if(!tituloPropiedad.tienePropietario())
                jugador.puedeComprarCasilla();
            else{
                tituloPropiedad.tramitarAlquiler(jugador);
            }
            
        }
    }
    
    public TituloPropiedad getTituloPropiedad(){
        return tituloPropiedad;
    }

    @Override
    public String toString(){
        String cadena = nombre + "\nPrecio: " + importe;
        
        return cadena;
    }
    
    private void informe(int actual, ArrayList<Jugador> todos) {
        if (super.jugadorCorrecto(actual, todos)) {
            Diario.getInstance().ocurreEvento(todos.get(actual).getNombre() + " ha caido en la calle " + this.toString());
        }
    }
    
}

File: Civitas_Ruby/lib/casilla_calle.rb
require_relative "titulo_propiedad.rb"
require_relative "sorpresa.rb"
require_relative "mazo_sorpresas"

module Civitas
  class Casilla_Calle < Casilla
    attr_reader :tituloPropiedad
    
    def initialize (titulo)
      super(titulo.nombre)
      @tituloPropiedad = titulo
      @importe = titulo.precioCompra
    end
    
    def recibeJugador(actual, todos)
      if(jugadorCorrecto(actual,todos))
        informe(actual,todos)
        jugador = Jugador.new(todos[actual])
        
        if(!@tituloPropiedad.tienePropietario)
          jugador.puedeComprarCasilla
        else
          @tituloPropiedad.tramitarAlquiler(todos[actual])
        end
      end
    end
    
    def to_s()
      cadena = "#{@nombre} Precio: #{@importe}"
      return cadena
    end
    
    def informe(actual, todos)
      if(jugadorCorrecto(actual,todos))
        Diario.instance.ocurre_evento(todos[actual].nombre.to_s + " ha caido en la casilla " + to_s)

      end
    end
    
  end
end